chrome/content/zotero/xpcom/attachments.js

```diff
@@ -332,6 +332,17 @@ Zotero.Attachments = new function(){
 					+ "'application/pdf' in Attachments.importFromURL()";
 				Zotero.debug(errString, 2);
 				Zotero.debug(sample, 3);
+				try {
+					if (tmpDir && tmpDir.exists()) {
+						tmpDir.remove(true);
+					}
+					if (destDir && destDir.exists()) {
+						destDir.remove(true);
+					}
+				}
+				catch (e) {
+					Zotero.debug(e, 1);
+				}
```

Comment:
Since this only relates to `tmpDir`, and `tmpDir` is passed to `Zotero.Utilities.Internal.saveURI`, I think it might make more sense to do the `tmpDir` clean up in that function.